tools/clang/include/clang/Frontend/CodeGenOptions.h

```diff
@@ -212,6 +213,21 @@ class CodeGenOptions : public CodeGenOptionsBase {
   std::vector<std::string> HLSLArguments;
   /// Helper for generating llvm bitcode for hlsl extensions.
   std::shared_ptr<hlsl::HLSLExtensionsCodegenHelper> HLSLExtensionsCodegen;
+  /// Helper for generating llvm bitcode for hlsl extensions.
+  std::shared_ptr<hlsl::options::OptimizationToggles> HLSLOptToggles;
```

Comment:
Why `std::shared_ptr`? This is the only owner so I think `std::unique_ptr` would be more appropriate

Reply:
`CodeGenOptions` gets many copies created, and we want them all to share the same instance of `HLSLOptToggles`, just like `HLSLExtensionsCodegen` above.

Reply:
Actually I'll add a comment about that. I used to wonder about the same thing.